Simplenote/TagListViewController.m

```diff
@@ -455,45 +455,27 @@ - (id)tableView:(NSTableView *)tableView objectValueForTableColumn:(NSTableColum
 
 - (NSView *)tableView:(NSTableView *)tableView viewForTableColumn:(NSTableColumn *)tableColumn row:(NSInteger)row
 {
-    if (row == kTopRow) {
-        // Just a spacer
-        return nil;
-    }
-    
-    NSString *cellId = @"TagCell";
-    
     if (row == kAllNotesRow) {
-        cellId = @"AllNotesCell";
-    } else if (row == kTrashRow) {
-        cellId = @"TrashCell";
-    } else if (row == kSeparatorRow) {
-        cellId = @"SeparatorCell";
+        return [self allNotesTableViewCell];
     }
-    
-    SPTagCellView *tagView = [self.tableView makeViewWithIdentifier:cellId owner:self];
-    [tagView.textField setDelegate:self];
-    [tagView setMouseInside:NO];
-    [tagView applyStyle];
-    
-    if (row == kAllNotesRow) {
-        tagView.textField.stringValue = NSLocalizedString(@"All Notes", @"Title of the view that displays all your notes");
-    } else if (row == kTrashRow) {
-        tagView.textField.stringValue = NSLocalizedString(@"Trash", @"Title of the view that displays all your deleted notes");
-    } else if (row == kSeparatorRow) {
-        tagView.textField.stringValue = @"";
-    } else {
-        Tag *tag = [self.tagArray objectAtIndex:row-kStartOfTagListRow];
-        tagView.textField.stringValue = tag.name;
+
+    if (row == kTrashRow) {
+        return [self trashTableViewCell];
     }
-    
-    return tagView;
+
+    if (row == kTagHeaderRow) {
```

Comment:
I'm liking that these conditionals aren't using `else` statements.  It's a little easier to read this way I think.

Reply:
Now that you mention it... that was one of the first comments Mike sent over, in a code review (circa 2013!).

If the first branch of the conditional ends up in a return, it really makes no sense to add an `else`, since... you avoid the indentation, and the `else` itself doesn't add anything to the logic.

🔥